lib/Jsrt/Jsrt.cpp

```diff
@@ -1367,102 +1367,199 @@ CHAKRA_API JsPreventExtension(_In_ JsValueRef object)
     });
 }
 
+static JsErrorCode InternalGetPropertyRecord(Js::ScriptContext * scriptContext, JsValueRef key, _Out_ const Js::PropertyRecord ** propertyRecord)
+{
```

Comment:
Is this deliberately weaker than being able to compute `x[y]`,  e.g. `var y = 2, var x = [1,2,3]; x[y]`? That is we deliberately do not support getting property records for numbers, null, true, false, undefined, or arbitrary objects which get converted to a string?

Reply:
We have an interface for number `JsGetIndexed...` (also works similarly) for the time being.. Once / if we ever  drop that interface, we may combine here. I was designing in a way that they are handled in the same place. Then I realized, consumer may have a better chance to know it hence no point wasting it here.